### PR TITLE
[dagit] Add Timeline to Instance Overview

### PR DESCRIPTION
**Pull request by hellendag**
## Summary

Add `RunTimeline` to the top of Instance Overview page. Made some changes to `RunTimeline` in the process.

- Use recent retrieved runs for timeline display. This is imperfect since it's just the last ten runs per job, not a true window.
- Use ten future ticks for enabled jobs. Again, imperfect because it's not windowed, though for every-minute jobs, it would be a little excessive to show e.g. the next hour of future ticks anyway.
- Modify batching logic in `RunTimeline` to batch based on visual overlap (pixel-based) rather than time overlap. This allows us to avoid cases where lots of runs are right up next to eachother but visually indistinguishable. Instead, they'll be batched with a number.
  - As a nice bonus for this, resizing the page will lead to nicely batching/unbatching runs based on the space available!
- Use pixel-level rendering instead of percentages when rendering run chunks, in an effort to have better spacing and consistency. I think it won't be perfect either way.
- To accomplish the bullet points above, start measuring the run chunk container, plus a ResizeObserver. This seems fairly performant even for the complicated storybook example.

## Test Plan

View Instance Overview, verify proper rendering of batches, hovercards, and scheduled tick items.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6117**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Comment by hellendag** [comment]
Screen Shot 2022-01-05 at 3 58 18 PM

**Review by prha — APPROVED** [review]

File: js_modules/dagit/packages/core/src/runs/RunTimeline.tsx [hunk 1]
```diff
@@ -21,61 +22,115 @@ const LABEL_WIDTH = 232;
...
+  jobs: JobMap;
   range: [number, number];
 }
 
 export const RunTimeline = (props: Props) => {
   const {jobs, range} = props;
+  const [width, setWidth] = React.useState<number | null>(null);
+  const observer = React.useRef<ResizeObserver | null>(null);
+
+  const containerRef = React.useCallback((node) => {
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/runs/RunTimeline.tsx:50]: I'd almost forgotten you could pass a function to `ref=`, this seems like a nice solution.

**Review by bengotow — COMMENTED** [review]
This looks great to me! I'm still a little concerned that we don't necessarily fetch enough data to fill the displayed time range though. It seems we should do a "fog of war" in the individual lanes so you know gaps that appear in the timeline aren't real gaps in execution.

What if we rendered the last few ticks / past runs fading out to `opacity:0` so if you see all 10 they gently disappear? Might let users know that more runs could exist past those?

**Pull request merged**